AirportDepartures.playground/Contents.swift

```diff
@@ -16,9 +16,68 @@ import UIKit
 //: e. Use a `String?` for the Terminal, since it may not be set yet (i.e.: waiting to arrive on time)
 //:
 //: f. Use a class to represent a `DepartureBoard` with a list of departure flights, and the current airport
-
-
-
+enum FlightStatus: String {
+    case scheduled = "Scheduled"
+    case canceled = "Canceled"
+    case enRoute = "En Route"
+    case landed = "Landed"
+    case boarding = "Boarding"
+    case departed = "Departed"
+}
+
+enum Timing: String {
+    case onTime = " - On Time"
+    case delayed = " - Delayed"
+    case early = " - Early"
+}
+
+struct Airport {
+    let name: String
+    let abbreviation: String
+    let location: String
+}
+
+struct Flight {
+    let airline: String
+    let number: String
+    let destination: Airport
+    var terminal: String?
+    var departureTime: String?
+    var status: (FlightStatus, Timing?)
+}
+
+class DepartureBoard {
+    var airport: Airport
+    var departures: [Flight]
+    
+    init(airport: Airport) {
+        self.airport = airport
+        self.departures = []
+    }
+    
+    func alertPassengers() {
+        for flight in departures {
+            if flight.terminal == nil {
+                print("Please see the nearest information desk about \(flight.airline) Flight \(flight.number) to \(flight.destination.name).")
+            } else {
+                switch flight.status.0 {
+                case .scheduled:
+                    print("\(flight.airline) Flight \(flight.number) to \(flight.destination.name) is scheduled to depart at \(flight.departureTime ?? "TBD") from Terminal: \(flight.terminal ?? "TBD")")
+                case .canceled:
+                    print("We're sorry \(flight.airline) Flight \(flight.number) to \(flight.destination.name) was canceled, here is a $500 voucher.")
+                case .enRoute:
+                    print("\(flight.airline) Flight \(flight.number) is en route, and scheduled to land at Terminal: \(flight.terminal ?? "TBD")")
+                case .landed:
+                    print("\(flight.airline) Flight \(flight.number) has landed at Terminal: \(flight.terminal ?? "TBD")")
+                case .boarding:
+                    print("\(flight.airline) Flight \(flight.number) is boarding, please head to Terminal: \(flight.terminal ?? "TBD") immediately. The doors are closing soon.")
+                case .departed:
+                    print("\(flight.airline) Flight \(flight.number) has departed from Terminal: \(flight.terminal ?? "unknown")")
+                }
+            }
+        }
+    }
```

Comment:
Nicely written alert function. The switch statement makes things very easy to read.